homeassistant/components/remote_calendar/config_flow.py

```diff
@@ -47,26 +60,57 @@ async def async_step_user(
             user_input[CONF_URL] = user_input[CONF_URL].replace(
                 "webcal://", "https://", 1
             )
+            _LOGGER.debug(
+                "Converted webcal:// to https:// URL: %s", user_input[CONF_URL]
+            )
         self._async_abort_entries_match({CONF_URL: user_input[CONF_URL]})
         client = get_async_client(self.hass)
+
         try:
+            _LOGGER.debug("Attempting to fetch calendar from URL")
             res = await get_calendar(client, user_input[CONF_URL])
+            _LOGGER.debug(
+                "Calendar fetch completed with status code: %s", res.status_code
+            )
+
+            if res.status_code == HTTPStatus.UNAUTHORIZED:
+                _LOGGER.warning(
+                    "Received 401 Unauthorized for URL. Response headers: %s",
+                    dict(res.headers),
+                )
+                # Store URL and calendar name for auth step
+                self.url = user_input[CONF_URL]  # pylint: disable=attribute-defined-outside-init
```

Comment:
Lets store `self.data = user_input` rather than the individual pieces, given we now have logic split across multiple steps it will be harder to manage.